compiler.go

```diff
@@ -29,6 +29,20 @@ type Code struct {
 	codeinfos []codeinfo
 }
 
+// RunNormalized runs the code with the variable values (which should be in the
+// same order as the given variables using [WithVariables]) and returns
+// a result iterator. Input numbers are normalized for CLI tests
+//
+// It is safe to call this method in goroutines, to reuse a compiled [*Code].
+// But for arguments, do not give values sharing same data between goroutines.
```

Comment:
Is this safe for our use-case? I thought our template data was passed around as pointers.

Reply:
TLDR, I updated the comment, it should be safe* to reuse values. normalizeNumbers, which we don't do in `Run` anymore, could mutate input values. It's unnecessary now (because of how we're passing values), and slow, and potentially destructive, so we don't do it in our usage.

*There are some jq functions, such as `setpath`, which are documented to mutate the underlying data. However, by the time jq gets data into something like `setpath`, we've either copied the data (e.g. `struct -> map[string]any`), or we don't support the operation on the type (no code in this library supports setting struct fields). I won't say it's impossible to mutate the data we pass in, but it's definitely not easy.